include/albatross/src/covariance_functions/traits.hpp

```diff
@@ -226,6 +226,25 @@ struct has_valid_variant_mean_caller<U, Caller, variant<A, Ts...>,
       has_valid_variant_mean_caller<U, Caller, variant<Ts...>>::value;
 };
 
+/*
```

Comment:
Does this belong in #357?

Reply:
It's only used in this PR, but it's true they address very similar issues.  I didn't want the templating:
```
  template <typename X>
  double _call_impl(const X &x, const X &y) const;
```
to capture any arbitrary type, only those which are "basic", it not composites of multiple types.